designs/cilium-configuration-policy.md

```diff
@@ -0,0 +1,119 @@
+# Configure Cilium CNI
+
+## Introduction
+
+**Problem:** The Cilium CNI plugin (built-in default for EKS Anywhere) supports defining and enforcement of network policies. Network policies determine how pods within a cluster are able to communicate with one another. Cilium accepts policy enforcement modes from the users to determine the allowed traffic between endpoints. Currently there is no way to configure Cilium launched by EKS Anywhere to use any mode apart from the default mode, which allows all traffic.
+
+### Goals and Objectives
+
+As an EKS Anywhere user:
+
+* I want to have the ability to configure Cilium's [supported policy enforcement modes](https://docs.cilium.io/en/v1.8/policy/intro/).
+* I want that, depending on the policy enforcement mode I select, EKS Anywhere should configure the network policies required for its core components to ensure that the cluster gets provisioned successfully.
+
+### Non-Goals/Future Work
+
+* Accepting network policies for user workloads through EKS Anywhere spec.
+
+
+## Overview of Solution
+
+With this feature, a user can launch Cilium with one of its supported policy enforcement modes: default, always, never. The default mode will be Cilium's default, which is also named `default`. The user can specify the mode through the EKS Anywhere cluster spec, and EKS Anywhere will use that while launching the Cilium Helm chart in the workload cluster.
+
+### Implementation details
+
+#### API Changes
+
+A new type will be added to accept the cni plugin name and additional options from the user:
+
+```go
+type CNIConfig struct {
```

Comment:
kinda want to know the possibility of using`CNIProviderRef` here. Especially when we want to support more CNIs